salt/haproxy/haproxy.manifest

```diff
@@ -24,7 +24,23 @@ spec:
       volumeMounts:
         - name: haproxy-cfg
           mountPath: /etc/haproxy
+{%- if "admin" in salt['grains.get']('roles', []) -%}
+        - name: velum-bundle-certificate
+          mountPath: /etc/pki/velum.pem
+          readOnly: True
+        - name: ca-certificate
+          mountPath: /etc/pki/ca.crt
+          readOnly: True
+{%- endif -%}
   volumes:
     - name: haproxy-cfg
       hostPath:
         path: /etc/haproxy
+{%- if "admin" in salt['grains.get']('roles', []) -%}
+    - name: velum-bundle-certificate
+      hostPath:
+        path: /etc/pki/velum.pem
+    - name: ca-certificate
+      hostPath:
+        path: /etc/pki/ca.crt
```

Comment:
Couldn't we just mount the `/etc/pki` directory? Mounting individual files is always problematic (the container will not see the change after a rename/rewrite/etc)

Reply:
If we mount the whole /etc/pki folder, then the container has access to all certificates. But - you are right - mounting files is problematic - we should probably re-evaluate where and how we store these files... something like this maybe?

* /etc/caasp/pki/haproxy/ca.crt (symlink if possible?)
* /etc/caasp/pki/haproxy/haproxy.crt
* /etc/caasp/pki/haproxy/haproxy.key
* /etc/caasp/pki/haproxy/haproxy-bundle.pem

But - that's a different issue..

Reply:
We could do the refactor you say @kiall, however the problem with bind mounts happens when the file is replaced in an automatic way. I don't know if this is the case at the moment with the certificates, something we can check. The downside is that even if it works, if salt on the future makes this replacement atomic we might be in a bad situation without even knowing it.

I think @kiall's suggestion is a good approach (if the symlink works -- even if it doesn't we could copy the CA cert with salt on each directory). If it's not much trouble, I'd propose to do that in a later refactor, as this changes are intrusive enough themselves...

Reply:
Why don't we create a certificates dir per service? `/etc/pki/haproxy` and so? With `cert.key` and `cert.crt` inside?

Reply:
(By the way, I think we should apply the same philosophy to all the files that are mounted in the manifests in the `containers-manifests` repo...)